--- lab.c	2025-11-25 20:10:34.912826796 +0300
+++ lab1.c	2025-11-25 20:10:45.375273447 +0300
@@ -117,10 +117,15 @@
     }
 }
 
-int main()
+int main(int argc, char *argv[])
 {
+    if (argc != 2)
+    {
+        printf("Parameters error!\n");
+        return 1;
+    }
     srand(seed);
-    int lab_size = 6;
+    int lab_size = atoi(argv[1]);
     int x_start = rand() % (lab_size - 2) + 1;
     int y_start = rand() % (lab_size - 2) + 1;
     char wall_c = '#';
